src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -881,7 +881,7 @@ public AnalyzerActions Append(AnalyzerActions otherActions, bool appendSymbolSta
             actions._operationBlockStartActions = _operationBlockStartActions.AddRange(otherActions._operationBlockStartActions);
             actions._operationBlockEndActions = _operationBlockEndActions.AddRange(otherActions._operationBlockEndActions);
             actions._operationBlockActions = _operationBlockActions.AddRange(otherActions._operationBlockActions);
-            actions._concurrent = actions._concurrent || otherActions.Concurrent;
+            actions._concurrent = _concurrent || otherActions.Concurrent;
```

Comment:
Consider adding a simple test to `DiagnosticAnalyzerTests`.
```C#
var a = new AnalyzerActions();
a.EnableConcurrentExecution();
var b = a.Append(new AnalyzerActions());
Assert.True(b.Concurrent);
```

Reply:
Appended the commit to include a test (no other changes).